lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -142,6 +142,22 @@ LogicalResult NamedTableOp::verify() {
   return success();
 }
 
+LogicalResult NamedTableOp::verify() {
+  llvm::ArrayRef<Attribute> fieldNames = getFieldNames().getValue();
+  auto tupleType = llvm::cast<TupleType>(getResult().getType());
+  return verifyNamedStruct(getOperation(), fieldNames, tupleType);
+}
+
+LogicalResult PlanRelOp::verifyRegions() {
+  if (!getFieldNames().has_value())
+    return success();
+
+  llvm::ArrayRef<Attribute> fieldNames = getFieldNames()->getValue();
+  auto yieldOp = llvm::cast<YieldOp>(getBody().front().getTerminator());
+  auto tupleType = llvm::cast<TupleType>(yieldOp.getValue().getType());
```

Comment:
And just to confirm, the return type of yield would have been verified before verifyRegions is called? (I don't recall the verifications order off the top of my head)

Reply:
I confirm via the [docs](https://mlir.llvm.org/docs/DefiningDialects/Operations/#verification-ordering):

> Note that the second phase [i.e., the region verifiers] will be run after the operations in the region are verified.